libraries/AP_BattMonitor/AP_BattMonitor_Analog.cpp

```diff
@@ -29,6 +29,11 @@ AP_BattMonitor_Analog::read()
     // get voltage
     _state.voltage = _volt_pin_analog_source->voltage_average() * _params._volt_multiplier;
 
+    // calculate cell count - WARNING this can be inaccurate if the cell detection voltage is not right or the battery is far from fully charged
+    if (!_state.cell_count) {
+        _state.cell_count = (uint8_t)(_state.voltage / _params._cell_detect_volt + 1);
```

Comment:
Potential division-by-zero here of `cell_detect_volt` here - yes, the user would have to work at it, but....

Reply:
I added a check for `cell_detect_volt <= 0`